### ISSUE TITLE
How do I use the encog data formats for audio classification with hidden markov models?

### ISSUE THREAD

**Issue by LSayn** [encog/encog-java-core#204]
I am trying to classify audio data with MFCC and an Hidden Markov Model. I am using the encog java library.

I have a working MFCC implementation and get 13 coefficients out of it.

I can't get it to work probably, I think my biggest problem is the understanding of the data format encog uses.

For training I have an array that holds all MFCC-vectors of the training data plus an array for information for supervised learning for each vector. Right now, I am simply tossing this in an BasicMlSequenceSet.

This "works", in terms of it doesn't throw an exception.

Classification works similiar, i put the MFCC vector in an BasicMLDataSet and call the probability-method of the hmm. It throws an ArrayIndexOutOfBoundsException here.

What am I doing wrong? I know I should normalize the features before using them, but how? Should I use a different data format here?

Sorry for the vagueness, but I am really stuck here and can't seem to get it to work. I am a little bit desperate here, so if anyone could help me out a little bit I would be more than thankful.

**Comment by LSayn** [comment]
In case that I am wrong here, I have posted the same question on Stackoverflow:
http://stackoverflow.com/questions/28801389/how-do-i-use-the-encog-data-formats-for-audio-classification-with-hidden-markov

**Comment by andrelopes1705** [comment]
LSayn, have you had any luck with this yet?
Im starting some toy-programs with audio and encog.

**Comment by LSayn** [comment]
@andrelopes1705 Nope. I haven't. I used Jahmm (just google it) and added the few things I needed.

**Comment by jeffheaton** [comment]
Encog does not have any support for reading audio directly.  However I will add an example of directly building a dataset of this type from scratch.

**Comment by true-meowmeow** [comment]
> Encog does not have any support for reading audio directly. However I will add an example of directly building a dataset of this type from scratch.

I am currently working with sound and have some difficulties. Is it possible to get a link to this example?
